### PR TITLE
[frontend] Profiles with fields & more

### PR DESCRIPTION
**Pull request by f0x52**
# Description

Redesigned the profile template, including showing profile fields, and re-organized header, bio, pinned posts etc.
More accessibility review for profile and status templates, resulting in more semantic use of `<details>/<summary>` and other tweaks. 

fixes https://github.com/superseriousbusiness/gotosocial/issues/1529
edit: fixes https://github.com/superseriousbusiness/gotosocial/issues/1663

and most of https://github.com/superseriousbusiness/gotosocial/issues/714 which could use an update / split into remaining issues

image

### CODE REVIEW

**Comment by f0x52** [comment]
looking at https://github.com/superseriousbusiness/gotosocial/issues/1663 and probably gonna sneak in a fix for that too

File: web/template/profile.tmpl [hunk 1]
```diff
@@ -18,80 +18,127 @@
...
+
+			<div class="bio">
+				{{ if .account.Note }}
+				{{emojify .account.Emojis (noescape .account.Note)}}
+				{{else}}
+				This GoToSocial user hasn't written a bio yet!
+				{{end}}
+			</div>
+
+			<div class="sr-only" role="group">
```
Comment (tsmethurst) [inline web/template/profile.tmpl:81]: Can you explain these two very similar divs? Not sure I understand what's going on here.

File: web/template/status.tmpl [hunk 2]
```diff
@@ -17,71 +17,114 @@
...
-			<a href="{{.URL}}"
-				 target="_blank"
-				 {{if .Description}}title="{{.Description}}"{{end}}
-				 data-pswp-width="{{.Meta.Original.Width}}px"
-				 data-pswp-height="{{.Meta.Original.Height}}px"
-				 {{if eq .Type "video"}}data-pswp-type="video"{{end}}
-				 data-cropped="true">
-				<img src="{{.PreviewURL}}" {{if .Description}}alt="{{.Description}}"{{end}} data-blurhash="{{.Blurhash}}"/>
-			</a>
+			<!-- {{if not .Description}}
```
Comment (tsmethurst) [inline web/template/status.tmpl:58]: This should probs be either uncommented or removed.

**Review by tsmethurst — COMMENTED** [review]
Great!!!

File: web/template/profile.tmpl (hunk 1 above)
Comment (f0x52) [reply web/template/profile.tmpl:84]: it's a bit of a hack to make that section read better, as visually you need the grid to have separate elements for each column entry, but that makes them all separately focussed/read. So the first div is only read for screen readers, second only shown visually

File: web/template/status.tmpl (hunk 2 above)
Comment (f0x52) [reply web/template/status.tmpl]: Removed for now. We might want to re-introduce the "this media is lacking a description" indicator at some point, but it doesn't really work well when there's multiple media on a post, and even worse on mobile where it obscures a large area

File: web/source/css/profile.css [hunk 3]
```diff
@@ -16,274 +16,252 @@
...
-	a img {
-		height: 5rem;
-	}
-}
-
-main {
-	background: transparent;
-	padding-top: 0;
+.page {
+	grid-template-columns: 1fr minmax(auto, 60rem) 1fr;
```
[the remark this answers is not in the document]
Comment (f0x52) [reply web/source/css/profile.css]: yep, added the comment

File: web/template/profile.tmpl (hunk 1 above)
Comment (tsmethurst) [reply web/template/profile.tmpl:84]: Aha okay, i suspected something like that but wasn't sure!

**Pull request merged**